src/EventStore.ClientAPI/ConnectionSettingsBuilder.cs

```diff
@@ -306,7 +306,25 @@ public class ConnectionSettingsBuilder {
 		/// <exception cref="ArgumentNullException">If <paramref name="clusterDns" /> is null or empty.</exception>
 		public ConnectionSettingsBuilder SetClusterDns(string clusterDns) {
 			Ensure.NotNullOrEmpty(clusterDns, "clusterDns");
-			_clusterDns = clusterDns;
+
+			var splits = clusterDns.Split(':');
+
+			switch (splits.Length) {
+				case 1:
+					_clusterDns = string.Format("{0}:2113", clusterDns);
```

Comment:
I don't think this port `2113` can be hard coded here.
It must (eventually) use `_gossipExternalHttpPort` but since that can be set (via `SetClusterGossipPort`) after this method is called, the `Build()` method should figure out the best port.

Reply:
It is not hard coded per se but rather defaulted if nothing is provided. You are supposed to declare the gossip port with the domain address with this patch.

Reply:
So this is a breaking change then?  Given what @shaan1337 said about about this code working before:

        var clusterSettings = ClusterSettings.Create()
            .DiscoverClusterViaDns()
            .SetClusterDns("clusterdnstest.eventstore.com")
            .SetClusterGossipPort(2113);